scrapy/downloadermiddlewares/robotstxt.py

```diff
@@ -48,23 +85,53 @@ def __init__(self, crawler: Crawler):
 
         # check if parser dependencies are met, this should throw an error otherwise.
         self._parserimpl.from_crawler(self.crawler, b"")
+        assert crawler.request_fingerprinter is not None
+        self._fingerprinter = crawler.request_fingerprinter
 
-    @classmethod
-    def from_crawler(cls, crawler: Crawler) -> Self:
-        return cls(crawler)
+        crawler.signals.connect(
+            self._start_request_returned, signal=signals.start_request_returned
+        )
+        crawler.signals.connect(
+            self._start_requests_exhausted, signal=signals.start_requests_exhausted
+        )
+
+    def _start_request_returned(self, request):
+        self._total_start_request_count += 1
```

Comment:
I thought about counting them on the engine and passing the count as a parameter through the signal, but it felt weird.